### DISCUSSION TITLE
如何才能创建umimax + electron项目呢？

### DISCUSSION THREAD

**Discussion by liaodalin19903** [umijs/umi#11099] [Q&A] [answered]
官方文档中有：
[创建UMI electron模板项目](https://umijs.org/docs/tutorials/getting-started#从-umijselectron-template-创建一个-electron-模板)

这里创建之后是umi + electron，请问下

1、如何才能创建umimax + electron项目呢？  
2、为何不出命令行工具创建 umimax + electron 项目呢？是没有必要对electron项目做umimax吗还是为何呢？

**Comment by fz6m** [comment, marked as answer]
umi 主要是普通的中后台 SPA 应用场景，比如 electron 这种 side 方案放到 umi 本身里，并没有能力持续维护，会造成历史债。

但社区也有人在做 electron + umi 一起使用的方案，所以将以外置模板的方式进行创建和使用（具体发展情况要看相关模板维护者的决策）。

如果你有相关的诉求，可以自己探究如何搭配使用，或者使用社区贡献的模板（如 [electron template](https://www.npmjs.com/package/@umijs/electron-template)），但一个成熟的 web + electron 结合开发需要更丰富的目录结构设计和更多的第三方高级工具包支持，具体情况可以自行探索。
